validate.go

```diff
@@ -41,6 +41,9 @@ func validateAndAdjustContainerResourceRequests(container *corev1.Container, set
 func validateAndAdjustContainerResourceLimit(container *corev1.Container, resourceName string, resourceConfig *ResourceConfiguration) (bool, error) {
```

Comment:
Unrelated with this PR, could you add some inline docs about what this function is doing? It took me a while to understand it's also mutating the `container` object given a input and understand the `bool` value returned is not about something being valid/invalid; but rather the `container` being mutated or not

Reply:
Also, can you update the inline documentation of `validateAndAdjustContainerResourceLimits` to reflect the behavior change introduced by this PR?